compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -2083,6 +2049,114 @@ static LogicalResult adjustTileSizesForUnPackOp(func::FuncOp entryPointFn,
+/// this case. The configuration is [[X, 0], [Y, 0], [0, Z], [0, 0]]. The `d0`
+/// is the common parallel dims for all the ops. The lowering config from
+/// `rootOperation` sets the tile sizes for `d0` and the rest of dims in
+/// reduction. But it does not have tile sizes for the rest of dims in
+/// elementwise op and pack ops. This method sets the vectorization tile sizes
+/// for other compute ops. E.g., [[X, 0], [Y, 0], [0, 0], [0, 4]] for the
+/// elementwise operations and [[X, 0], [Y, 0], [0, 0], [0, 16]] for the pack
+/// op.
+static void setLoweringConfigForComputeOps(func::FuncOp entryPointFn,
+                                           ArrayRef<Operation *> computeOps,
+                                           Operation *rootOperation) {
+  auto ctx = entryPointFn.getContext();
+  auto rootLoweringConfig = getLoweringConfig(rootOperation);
+  auto distTileSizes = rootLoweringConfig.getTileSizeVals(0);
+  auto tileAndFuseSizes = rootLoweringConfig.getTileSizeVals(1);
+  auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(entryPointFn);
+  auto targetMLTransInfo =
+      TargetMLTransformInfo::getTargetMLTransformInfo(targetAttr);
+  for (auto op : computeOps) {
+    // The lowering config is already set on rootOperation, so we skip it.
+    if (op == rootOperation)
+      continue;
+
+    int numLoops = cast<TilingInterface>(op).getLoopIteratorTypes().size();
+    SmallVector<int64_t> zeros(numLoops, 0);
+    TileSizesListType tileSizesList = {distTileSizes, tileAndFuseSizes};
+    TypeSwitch<Operation *>(op)
+        .Case<tensor::PackOp>([&](auto packOp) {
+          SmallVector<int64_t> vecTileSizes =
+              getPackVectorTileSizes(entryPointFn, packOp);
+          tileSizesList.push_back(zeros); // tensor.pack op does not have
+                                          // reduction loops.
+          tileSizesList.push_back(vecTileSizes);
+        })
+        .Case<linalg::GenericOp>([&](auto genericOp) {
+          auto vecPreProcStrategy = getVectorPreProcStrategy(genericOp);
+          auto linalgOpInfo = LinalgOpInfo(genericOp);
+          int64_t vecSize = getNativeVectorSizeInBytes(entryPointFn) / 4;
+          SmallVector<int64_t> vecTileSizes = getMinTilingSizesForEachDim(
+              entryPointFn, genericOp, linalgOpInfo, targetMLTransInfo);
+          for (auto &i : vecTileSizes) {
+            i = roundUpToPow2(std::min(i, vecSize),
+                              vecPreProcStrategy ==
+                                  VectorPreProcStrategy::Masking);
+          }
+          // If the dimension is already tiled, we don't tile it again.
+          if (!isRISCV(targetAttr)) {
+            for (auto i :
+                 llvm::seq<int64_t>(0, std::min(tileAndFuseSizes.size(),
+                                                vecTileSizes.size()))) {
+              if (tileAndFuseSizes[i])
+                vecTileSizes[i] = 0;
+            }
+          }
```

Comment:
Could you please add a note/example about why this is needed? Otherwise, it will be difficult to get rid of it/investigate this further.

Reply:
done